cpp_faster_fifo/cpp_lib/faster_fifo.cpp

```diff
@@ -37,8 +37,17 @@ struct Queue {
         return max_size_bytes;
     }
 
-    [[nodiscard]] bool can_fit(size_t data_size) const {
-        return size + data_size <= max_size_bytes;
+    [[nodiscard]] size_t get_max_size() const {
+        return maxsize;
+    }
+
+    [[nodiscard]] bool can_fit(size_t data_size, size_t additional_size) const {
+        bool cond_size;
```

Comment:
small suggestion to just assign expressions directly to bool variables without creating uninitialized booleans